### DISCUSSION TITLE
<br> tag equivelent?

### DISCUSSION THREAD

**Discussion by DataStream3** [AvaloniaUI/Avalonia#6887] [Q&A]
This is probably a stupid question, but is there an HTML br tag equivalent in Avalonia? I tried looking through the documentation but I couldn't see anything similar. Thank you in advance.

**Comment by robloo** [comment]
I'm assuming you are needing this in XAML. XAML should support line breaks using the same code as WPF:

```xaml
<TextBlock>Some Text <LineBreak/> Even More</TextBlock>
```

[Here is another example](https://github.com/AvaloniaUI/Avalonia/issues/6765#issue-1030801552) if `<LineBreak/>` doesn't work yet.
```xaml
<ToolTip.Tip>
  <TextBlock>
    Returns a Domino build given a:&#x0a;
    - Build Number (e.g. 16.0.11217.10000)&#x0a;&#x0a;
    Returns the last 2 Domino builds given a:&#x0a;
    - Leave Blank for Daily Builds
  </TextBlock>
</ToolTip.Tip>
```

There is a big caveat though: Whitespace handling modes in Avalonia are poorly supported. While whitespace is generally just trimmed, it is somewhat undefined.

 * https://github.com/AvaloniaUI/Avalonia/issues/2619
 * https://github.com/AvaloniaUI/Avalonia/issues/6765

If in code-behind, obviously just use `Environment.NewLine`.

**Comment by maxkatz6** [comment]
Really depends on what you need to do. Xaml is not Html. And xaml app is not a document (while html app technically is). We don't use BR or anything like this to create application layout, instead we are using Grid, StackPanel, DockPanel and other kinds of panels including ItemsPanel (and based on it ListBox). 
On other hand, if you only want to format text, @robloo already answered this part. Though it's also often to format text in the view model (c# code) or resources files (resx file per language).
